src/main/java/gregtech/integration/jei/recipe/RecipeMapCategory.java

```diff
@@ -57,12 +56,13 @@ public RecipeMapCategory(RecipeMap<?> recipeMap, IGuiHelper guiHelper) {
                 (importItems = new ItemStackHandler(recipeMap.getMaxInputs())),
                 (exportItems = new ItemStackHandler(recipeMap.getMaxOutputs())),
                 (importFluids = new FluidTankList(false, importFluidTanks)),
-                (exportFluids = new FluidTankList(false, exportFluidTanks)),
-                (recipeMap.getMaxOutputs() >= 6 || recipeMap.getMaxInputs() >= 6 ||
-                        recipeMap.getMaxFluidOutputs() >= 6 || recipeMap.getMaxFluidInputs() >= 6) ? FONT_HEIGHT : 0
+                (exportFluids = new FluidTankList(false, exportFluidTanks)), 0
         ).build(new BlankUIHolder(), Minecraft.getMinecraft().player);
         this.modularUI.initWidgets();
-        this.backgroundDrawable = guiHelper.createBlankDrawable(modularUI.getWidth(), modularUI.getHeight() * 2 / 3);
+        this.backgroundDrawable = guiHelper.createBlankDrawable(modularUI.getWidth(), modularUI.getHeight() * 2 / 3 + (FONT_HEIGHT *
+                (((recipeMap.getMaxOutputs() >= 6 || recipeMap.getMaxInputs() >= 6 ||
+                        recipeMap.getMaxFluidOutputs() >= 6 || recipeMap.getMaxFluidInputs() >= 6) && recipeMap.getRecipeList().get(0).getPropertyValues().size() > 0)
```

Comment:
It would be cleaner and easier to read by moving the input amount checking to a new, private, method